### PR TITLE
Do not feed any empty logic into scheduling

### PR DESCRIPTION
**Pull request by gezalore**
I would just push this but I still get "No space left on device" in the CI on my fork, so double checking.

### CODE REVIEW

File: src/V3ActiveTop.cpp [hunk 1]
```diff
@@ -77,6 +77,21 @@ class ActiveTopVisitor final : public VNVisitor {
             VL_DO_DANGLING(nodep->unlinkFrBack()->deleteTree(), nodep);
             return;
         }
+
+        // Delete empty procedures
+        for (AstNode *stmtp = nodep->stmtsp(), *nextp; stmtp; stmtp = nextp) {
+            nextp = stmtp->nextp();
+            if (AstNodeProcedure* const procp = VN_CAST(stmtp, NodeProcedure)) {
+                if (!procp->stmtsp()) VL_DO_DANGLING(procp->unlinkFrBack()->deleteTree(), stmtp);
```
Comment (wsnyder) [inline src/V3ActiveTop.cpp:85]:
```suggestion
                if (!procp->stmtsp()) VL_DO_DANGLING(pushDeletep(procp->unlinkFrBack()), stmtp);
```

Makes it easier to debug as new nodes won't get new'ed at old proc's address.  Similar below.

File: src/V3OrderCFuncEmitter.h [hunk 2]
```diff
@@ -116,7 +116,7 @@ class V3OrderCFuncEmitter final {
         AstNode* const headp = [&]() -> AstNode* {
             if (!procp) return logicp;  // Not a procedure, handle as a unit
             AstNode* const stmtsp = procp->stmtsp();
-            if (stmtsp) stmtsp->unlinkFrBackWithNext();
+            stmtsp->unlinkFrBackWithNext();
```
Comment (wsnyder) [inline src/V3OrderCFuncEmitter.h:119]:
```suggestion
            UASSERT_OBJ(stmtp, procp, "Empty processes should have deleted earlier");
            stmtsp->unlinkFrBackWithNext();
```

**Review by wsnyder — COMMENTED** [review]
I was thinking we were just on edge of disk space but I'm now thinking github changed something.  I might start deleting files on pass.

Anyhow other feedback on this pull, then feel free to merge at will.

**Comment by gezalore** [comment]
@wsnyder might this be the CI issuue? https://github.com/actions/runner/issues/3207 There is a workaround mentioned in there.

**Comment by wsnyder** [comment]
Please merge in master to get CI resolution (thanks!), and assuming passes feel free to squash merge.

**Pull request merged**